Client/src/Validation/validation.js

```diff
@@ -128,10 +128,57 @@ const maintenanceWindowValidation = joi.object({
 	monitors: joi.array().min(1),
 });
 
+const advancedSettingsValidation = joi.object({
+	apiBaseUrl: joi.string().uri({ allowRelative: true }).trim().messages({
+		"string.empty": "API base url is required.",
+		"string.uri": "The URL you provided is not valid.",
+	}),
+	logLevel: joi.string().valid("debug", "none", "error", "warn").allow(""),
+	systemEmailHost: joi.string().allow(""),
+	systemEmailPort: joi.string().allow(""),
+	systemEmailAddress: joi.string().allow(""),
+	systemEmailPassword: joi.string().allow(""),
+	jwtTTL: joi
+		.string()
+		.trim()
+		.messages({
+			"string.empty": "JWT TTL is required.",
+		})
+		.custom((value, helpers) => {
+			// test against regx start with a number followed by number or .,
+			//ends with d or h
+			const regex = /^[1-9]+\d*(.)*(d|h)$/;
+			const found = value.match(regex);
+			if (!found) {
+				return helpers.message(
+					"JWT TTL should start with a non zero number and ends with 'd' or 'h'."
+				);
+			}
+			return value;
+		}),
+	dbType: joi.string().trim().messages({
+		"string.empty": "DB type is required.",
+	}),
+	redisHost: joi.string().trim().messages({
+		"string.empty": "Redis host is required.",
+	}),
+	redisPort: joi
+		.string()
+		.allow("")
+		.custom((value, helpers) => {
+			if (value && Number.isNaN(parseInt(value))) {
```

Comment:
:+1:

Reply:
Just to confirm , 
1. we will need to add a dropdown for day and hour ?
2. Add onBlur for input components
3. Fix the regex to allow only numbers ending with a 'd' or 'h' and no fraction numbers(remove the dot)

Reply:
1 - Yes, and validate on that step instead of onChange
2- Since we are splitting the input into a number and a time unit, we could split validation as well. This way you could validate if is a number, and the other would use the same sort of validation we used for the other select.
3 - Yes, I think so. Gorkem sent a message on discord with the figma image

Reply:
resoved in : https://github.com/bluewave-labs/bluewave-uptime/pull/1020/commits/a4344b89a23692d4d7cb2a534b90d99044d5ae24
https://github.com/bluewave-labs/bluewave-uptime/pull/1020/commits/57edff36c9920fc569f24c7c11dab21f81704b12